models/users.js

```diff
@@ -576,6 +576,23 @@ const fetchAllUsers = async () => {
   return users;
 };
 
+const updateRoles = async (userData, newRoles) => {
+  try {
+    const roles = { ...userData.roles };
+    const newRolesArray = Object.entries(newRoles);
+    if (roles[newRolesArray[0][0]] === newRolesArray[0][1]) return { isRoleUpdated: false };
+    const updateRoles = { ...roles, ...newRoles };
+    await userModel.doc(userData.id).update({
+      updateRoles,
+    });
+
+    return { isRoleUpdated: true };
+  } catch (err) {
+    logger.error("Error while updating roles for a user", err);
+    throw err;
+  }
+};
+
```

Comment:
Could we use the pre-created function named updateUser.

because resource being updated is user

Reply:
It is not `updateUser` function -> It is `addOrUpdate` function in model/users.js
If we add `updateRoles` function logic in `addOrUpdate` function because of this reason that roles is one of the resource of users

But after adding the logic of updatRoles function in addOrUpdate function then may be it create problem or break our code and  addOrUpdate function code/logic will not be understable because adding all logic in the same function.

Reply:
The `addOrUpdateUser` function is highly capable of changing and updating user information. It saves us from the hassle of creating numerous new functions for every minor update. Just imagine, if tomorrow I want to update my first name, would it be necessary to create a new function specifically called `updateFirstName`? And then another function for `updateLastName`, followed by functions for `updateDob`, `updateLinkedin`, and so forth? By utilizing the versatile `addOrUpdateUser` function, we can conveniently handle all these updates in a single place.

Reply:
Done can you check my latest commit